Environment_Engine/Query/Orientation.cs

```diff
@@ -42,17 +43,64 @@ public static partial class Query
 
         [Description("Returns the orientation of a given environmental object")]
         [Input("environmentObject", "Any object implementing the IEnvironmentObject interface that can have its orientation queried")]
+        [Input("northAngle", "The angle in degrees for north. Default is 90.0.")]
+        [Input("azimuthAngle", "Set to true to return the azimuth angle from north instead of the angle between north and the object normal. Default is false.")]
         [Output("orientation", "The orientation of the Environment Object")]
-        public static double Orientation(this IEnvironmentObject environmentObject)
+        public static double Orientation(this IEnvironmentObject environmentObject, double northAngle = 90.0, bool azimuthAngle = false)
         {
+            double northAngleRadians = northAngle * (Math.PI / 180);
+            Vector northVector = BH.Engine.Geometry.Create.Vector((double)Math.Cos(northAngleRadians), (double)Math.Sin(northAngleRadians));
             Polyline pLine = environmentObject.Polyline();
-
             List<Point> pts = pLine.DiscontinuityPoints();
-            Plane plane = BH.Engine.Geometry.Create.Plane(pts[0], pts[1], pts[2]); //Some protection on this needed maybe?
+            Plane plane = BH.Engine.Geometry.Create.Plane(pts[0], pts[1], pts[2]);
+            Vector normal = plane.Normal;
+            Vector up = BH.Engine.Geometry.Create.Vector(0, 0, 1);
+            Vector down = BH.Engine.Geometry.Create.Vector(0, 0, -1);
+            Vector right = BH.Engine.Geometry.Create.Vector(1, 0, 0);
+
+
+            if (normal == up || normal == down)
+            {
+                BH.Engine.Reflection.Compute.RecordError("This method cannot successfully evaluate the orientation");
+                return (-1);
+            }
+
+            else if (normal.X > 0 && northAngle == 90.0 && azimuthAngle == true)
```

Comment:
If the element we're evaluating is a boolean, like `azimuthAngle`, we don't need to do an `== true` - we can just do `if(azimuthAngle)` for example. If we want to evaluate whether it is false, we can use the `!` operator (sometimes referred to as the bang operator, or not operator) to indicate a `not` value. For example, `!azimuthAngle` will ask whether the variable is not true (i.e. false). This is the accepted convention for boolean evaluations rather than `== true` or `== false` :smile:

So the change needed for this line, and all the other lines below it, is to simply remove `== true` from the statement :smile: